velox/connectors/hive/HiveConnector.h

```diff
@@ -209,7 +210,7 @@ class HiveDataSource : public DataSource {
 
   std::shared_ptr<HiveConnectorSplit> split_;
   FileHandleFactory* fileHandleFactory_;
-  dwio::common::ReaderOptions readerOpts_;
+  std::shared_ptr<dwio::common::ReaderOptions> readerOpts_;
```

Comment:
This is what I was taught. Is there a downside? I don't think raw pointers should be used unless there is a VERY good reason.

Reply:
Why not leave it as it was before `dwio::common::ReaderOptions readerOpts_`?

Reply:
Do you mean pass it and store it as a "const dwio::common::ReaderOptions&"? That will be a more difficult implementation as its not created in one single location in the code. shared pointers means ownership/lifetime is based on need and not on a fixed single owner. If you insist, of course I will change it all but I do disagree.

Reply:
No, I mean to keep it as an object `dwio::common::ReaderOptions readerOpts_` of `HiveDataSource` like it was before. 
I see that you changed the API of `CachedBufferedInput` to accept `ReaderOptions` as well.
 `HiveDataSource` and `CachedBufferedInput` can probably share a copy with this change. This change is fine then.

Reply:
I am not sure I agree that this should be a shared_ptr, rather than a simple object.  Can you please explain the logic in more detail again as to why?  A shared means that a change in one location will change all uses, which is not necessarily what you would expect in this case.

Also the way the set methods of the reader are set up could cause problems of lifetime. What does it mean if I have std::shared_ptr<ReaderOptions> opt1; auto opt2 = opt1->SetMaxDistance().  I just made a copy of opt2 from opt1.  

I think we are mixing paradigms here and think this should be left as a ReaderOptions and not a shared_ptr